lib/checkother.cpp

```diff
@@ -485,6 +485,11 @@ void CheckOther::checkRedundantAssignment()
                 else
                     start = tok->findExpressionStartEndTokens().second->next();
 
+                // Is this a variable declaration with initialization?
+                const Token* defineAndInitTok = start->tokAt(-3);
+                if(defineAndInitTok->str()==";" && defineAndInitTok->linenr() == start->linenr())
```

Comment:
I would suggest to make this condition more explicit. It's easier to read what you expect if you'd write:

    if (Token::Match(defineAndInitTok, "; %var% =") && defineAndInitTok->linenr() == start->linenr())

Reply:
Ok I have to get used to this notation/checking...will change the code in this way.
Regarding Appveyor/Travis check: I updated the posix.c check because with the update redundantAssigment check the suppression isn't needed anymore.
The check update simply tries to seperate variable defintion with initialization with pure redundant assignments.